### ISSUE TITLE
[BUG] Some Series show up as unwatched from Trakt Collection

### ISSUE THREAD

**Issue by Mogulman12** [umbrellaplug/umbrellaplug.github.io#305]
### Bug

Intermittently, I've noticed some series/seasons don't show up as watched in Umbrella with Trakt integration.   This has been happening for quite a while (6-12 months).   The same series on other Kodi plug-ins do show up as watched.   
Latest example of this is the show.  Dr. Pimple Popper.   There are 9 seasons.  All of the episodes and seasons are watched in Trakt.  Verified on the Trakt website (100% watched).     When I go into Umbrella, and select Dr. Pimple Popper, and go to Season 1.  It shows all the episodes watched, but the season is not watched.     

I've reinstalled Umbrella on different devices and get the same behavior.  I've tried resyncing Trakt, reauthorizing trakt.    

In the attached log, I went to season 1 and marked "watched" it Trakt.   Then I went into Season 2, and marked watched in Trakt.  Yet I still see the season as unwatched even though all the episodes are watched.  I didn't see any unwatched specials either.

https://paste.kodi.tv/ocapihubag
[2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_watchedProgress() Line # :1531
                       msg : Trakt Progress List Sync Update...(local db latest "list_cached_at" = 1690861278, trakt api latest "progress_activity" = 1690902590)
[2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_user_lists() Line # :1587
                       msg : Trakt User Lists Sync Update...(local db latest "lists_updatedat" = 72000, trakt api latest "lists_updatedat" = 1493688601)
[2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_liked_lists() Line # :1614
                       msg : Trakt Liked Lists Sync Update...(local db latest "liked_at" = 72000, trakt api latest "liked_at" = 1493688601)
[2023-08-01 09:25:38] [ plugin.video.umbrella: WARNING ]: getTrakt() (404:NOT FOUND): URL=(https://api.trakt.tv/shows/tt8564914/progress/watched?specials=false&hidden=false): 
[2023-08-01 09:25:50] [ plugin.video.umbrella: WARNING ]: getTrakt() (404:NOT FOUND): URL=(https://api.trakt.tv/shows/tt8564914/progress/watched?specials=false&hidden=false): 

### Expected Result

I'd expect the seasons of this show to show as watched if all episodes are watched.   If all seasons are watched, then the series should show as watched.

### Steps To Reproduce

Add Dr. Pimple Popper to a collection.
Mark season 1 as watched using Trakt
See if you still see season 1 as being watched or unwatched even though all the episodes are watched.
If I am within the series Dr. Pimple Popper within Umbrella, and use the left panel to view only unwatched, it shouldn't show season 1, but it doesn.

### Kodi Version, Type of Device, Logs, Screenshots, and Additional Info

Kodi 20.2 (but this has been happening for a long time).  Latest version of umbrella.

https://paste.kodi.tv/ocapihubag

 [2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_watchedProgress() Line # :1531
                       msg : Trakt Progress List Sync Update...(local db latest "list_cached_at" = 1690861278, trakt api latest "progress_activity" = 1690902590)
[2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_user_lists() Line # :1587
                       msg : Trakt User Lists Sync Update...(local db latest "lists_updatedat" = 72000, trakt api latest "lists_updatedat" = 1493688601)
[2023-08-01 09:25:18] [ plugin.video.umbrella: DEBUG ]: From func name: resources.lib.modules.trakt.sync_liked_lists() Line # :1614
                       msg : Trakt Liked Lists Sync Update...(local db latest "liked_at" = 72000, trakt api latest "liked_at" = 1493688601)
[2023-08-01 09:25:38] [ plugin.video.umbrella: WARNING ]: getTrakt() (404:NOT FOUND): URL=(https://api.trakt.tv/shows/tt8564914/progress/watched?specials=false&hidden=false): 
[2023-08-01 09:25:50] [ plugin.video.umbrella: WARNING ]: getTrakt() (404:NOT FOUND): URL=(https://api.trakt.tv/shows/tt8564914/progress/watched?specials=false&hidden=false): 

### Checklist

- [X] I have included logs and the version of Kodi I am using..
- [X] I have made a separate issue for each bug that I am reporting.
- [X] I have verified this is not a setting.

**Comment by Mogulman12** [comment]
Seems like maybe the IMDB ID is incorrect.   If I search for Dr. Pimple Popper in IMDB, I see tt6045142.   Maybe it changed or something?  Trakt is using the TT6045142 id not the other one?

Fen uses TMDB ID=80991 and TVDB ID=340489.  Both of those seem to be correct.

**Comment by Mogulman12** [comment]
Errr...If I do a trakt search I can find it using tt8564914.   Not sure why it's failing using the api.  Or maybe something else is not working.  Tried to scour through the Kodi logs and Umbrella logs, but its hard to follow.

**Comment by umbrellaplug** [comment]
TMDB ID is: 80991
Trakt ID appears to be: 126690
IMDB ID is: tt6045142

Not sure if there's something weird with that one or not. I will try to follow your steps to re-produce the issue and see if I can track it down.

**Comment by umbrellaplug** [comment]
Are you using any skins?

**Comment by Mogulman12** [comment]
Default (no skins).

**Comment by umbrellaplug** [comment]
Please let me know if this is still happening in the latest version.

**Comment by Mogulman12** [comment]
Still happening on 6.6.28.  Here are some pics.  You can see all the episodes in season 1 are watched.  But the season shows as unwatched.   If I try and mark watched, an episode in season 1, I get the error above.  
![20230821_094500](https://github.com/umbrellaplug/umbrellaplug.github.io/assets/123271671/8319d318-b6dc-4d31-bdb4-948c67bab850)
![20230821_094452](https://github.com/umbrellaplug/umbrellaplug.github.io/assets/123271671/647e5730-f5dd-4cb0-b9ce-0fd8bdab1ecd)

**Comment by umbrellaplug** [comment]
Can you try the latest test version? I've made changes that should resolve your issue. You likely will need to clear all cache.

**Comment by Mogulman12** [comment]
Where is the test repo located at?

**Comment by umbrellaplug** [comment]
https://umbrellaplug.github.io/umbrellatest

**Comment by Mogulman12** [comment]
Seems to be working.  Installed new version.  Cleared cache and synced with trakt.   Tried marking watched and unwatched and also waiting 45 mins to see if an automated sync would cause problems.  All seems good.

**Comment by Mogulman12** [comment]
Just need to put it in the main build

**Comment by umbrellaplug** [comment]
Thanks for testing that out and being patient for the long wait to look at this. It's caused by an extremely hard to find bug where season imdb ids were replacing show imdb ids in rare instances. I'm currently working on a couple of features once I complete those I will push this fix with those features in the next update.
